### DISCUSSION TITLE
Different run_every value and gap results

### DISCUSSION THREAD

**Discussion by rschirin** [jertel/elastalert2#1034] [Q&A] [answered]
Hi,
I was trying to overwrite the global `run_every` key (1 minute) specifying it inside rule's yaml.
This is my scenario:
```
rule_type: spike
timeframe: 7d
run_every: 4h
```

Unfortunately, if I use this configuration I will have an investigation gap since the timeframe between 2 executions will not be checked.
i.e. 
first run: from 2022-11-21 07:30 UTC to 2022-11-28 07:30 UTC
second run: from 2022-11-21 11:30 UTC to 2022-11-28 11:30 UTC

Is there a way to avoid this?

**Comment by jertel** [comment, marked as answer]
I'm struggling to see the gap you're reporting. Where is the 4h gap falling in the timeframe? Beginning or end?

**Reply by rschirin** [reply]
sorry, my fault since I used a wrong example scenario. 
Issue is faced when the `run_every` key value is greater than `timeframe` key value.
i.e.
```
rule_type: spike
timeframe: 1h
run_every: 2h
```

**Reply by jertel** [reply]
It seems like the simple solution is to change the values to ensure there is no gap.

**Reply by rschirin** [reply]
Yeah, I know. Anyway there is a tricky constraint with business policy. I saw there is also query_delay key. It should be my fix.
